Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -13,6 +13,7 @@ def possible_javas
     javas << Pathname.new(ENV["JAVA_HOME"])/"bin/java" if ENV["JAVA_HOME"]
     javas << java_home_cmd
     javas << which("java")
+    javas.delete(Pathname.new("/usr/bin/java")) # /usr/bin/java is a stub on macOS
```

Comment:
Seems like `preferred_java` will check the truthiness of each member of this array, but I agree that it's cleaner this way.